tianshou/data/batch.py

```diff
@@ -263,18 +264,37 @@ def __init__(self,
                  **kwargs) -> None:
         if copy:
             batch_dict = deepcopy(batch_dict)
-        if _is_batch_set(batch_dict):
-            self.stack_(batch_dict)
-        elif isinstance(batch_dict, (dict, Batch)):
-            for k, v in batch_dict.items():
-                if isinstance(v, dict) or _is_batch_set(v):
-                    self.__dict__[k] = Batch(v)
-                else:
-                    if isinstance(v, list):
-                        v = np.array(v)
-                        if not issubclass(v.dtype.type, (np.bool_, np.number)):
-                            v = v.astype(np.object)
-                    self.__dict__[k] = v
+        if batch_dict is not None:
+            if isinstance(batch_dict, (dict, Batch)):
+                for k, v in batch_dict.items():
+                    if isinstance(v, (list, tuple, np.ndarray)):
+                        v_ = None
+                        if isinstance(v, np.ndarray):
+                            v_ = v
+                        elif all(isinstance(e, torch.Tensor) for e in v):
```

Comment:
Previous code (like db0e2e5cd21071fa6056a9f3e3c84b540bf80bf6 ) fails on ``Batch(a=[torch.ones(3, 3), torch.ones(3, 3)])``. This commit fixes it.

Reply:
Indeed, nice catch. You should add this test to the unit tests.